hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/operations/ReadManyOperation.java

```diff
@@ -53,20 +53,20 @@ public ReadManyOperation(String name, long startSequence, int minSize, int maxSi
 
     @Override
     public void beforeRun() {
-        RingbufferContainer ringbuffer = getRingBufferContainer();
-        ringbuffer.checkBlockableReadSequence(startSequence);
     }
 
     @Override
     public boolean shouldWait() {
         if (resultSet == null) {
-            resultSet = new ReadResultSetImpl<O, O>(minSize, maxSize, getNodeEngine().getSerializationService(), filter);
+            resultSet = new ReadResultSetImpl<>(minSize, maxSize, getNodeEngine().getSerializationService(), filter);
             sequence = startSequence;
         }
 
         RingbufferContainer ringbuffer = getRingBufferContainer();
+        sequence = ringbuffer.clampReadSequenceToBounds(sequence);
+
         if (minSize == 0) {
-            if (!ringbuffer.shouldWait(sequence)) {
+            if (sequence < ringbuffer.tailSequence() + 1) {
```

Comment:
Can we leave the previous condition? After clamping, we should be fine and it should no longer throw a exception.

Reply:
Ok.